src/summaries/chapter5.yaml

```diff
@@ -0,0 +1,136 @@
+- title: Understanding Encapsulation in Solidity
+  shortTitle: Encapsulation
+  key: encapsulation
+  details: |
+    It refers to the bundling of data, along with the methods that operate on that data, into a single unit.
+    It refers to the mechanism of manipulating the scope of variables, i.e., restricting the variable's access outside the scope.
+  
+    Syntax
+      ```
+      solidity
+      // Unlike in Java,
+      // the data type comes before the visibility modifier in Solidity
+      uint public num;
+      // <data type> <visibility> <variale name>;
+      ```
+  
+    - Public
+      * These objects can be accessed internally and externally as well as via messages also.
+      * External methods can inherit public elements.
+      * A getter function is generated automatically when a public variable is created.
+      * However, no setter function is created, implying that the external function can not modify the variable.
+  
+    - Internal
+      * As the name suggests, Internal objects can be accessed by internal or derived methods but cannot be accessed by external.
+      * Only a base contract and a derived contract have their access.
+  
+    - Private
+      * Private objects can only be accessed internally from the current contract instances.
+      * They cannot be accessed by derived methods also.
+  
+    - External
+      * These objects can be accessed externally but not internally, i.e., current contract instances cannot access it.
+      * They cannot be inherited.
+    
+    State Variable visibility
```

Comment:
Also mention one or two use cases i.e. 

* Use public for variables when you want the to be visible outside

and so on

Reply:
Use cases are mentioned in the detail section